src/main/code.ts

```diff
@@ -5,125 +5,422 @@
 			}
 
+		}
+	}
+
+	// Getting the numbers of columns of the selected table and comparing it to the data
+	var columnLength = selectedFrame.children.length
+	if (columnLength != columnLengthData) {
+		if (resizable) {
+			updateColumnCount(selectedFrame, columnLength, columnLengthData)
+			columnLength = columnLengthData
 		} else {
-		  getFirstChildOfTypeText(newNode).textAlignHorizontal = 'LEFT' // fails on "'" character
+			// Check for column selection
+			var selectionIsColumn = true
+			for (var i = 0; i < selectedFrame.children.length; i++) {
+				if (selectedFrame.children[i].type != "INSTANCE") {
+					selectionIsColumn = false
+				}
+			}
+
+			if (selectionIsColumn) {
+				//column selected
+				for (var i = 0; i < selectedFrame.children.length; i++) {
+					var textNode = getFirstChildOfTypeText(selectedFrame.children[i] as SceneNode)
+					if (textNode != null) {
+						textNodes.push(textNode)
+					}
+				}
+
+				if (columnLengthData == 1 && rowLengthData == textNodes.length) {
+					insertData(textNodes, cellsData, right_align_numbers)
+				} else {
+					figma.closePlugin("🛑 This seems to be a column with a different cell count than given by your data");
+				}
+
+				return
+			} else {
+				figma.closePlugin("🛑 Your given data has not the same column count as your selected table");
+				return
+			}
 		}
-		
-		if (wrapInAutoLayout) {
-		  columnAutoLayouts[j].appendChild(newNode)
-		  columnAutoLayouts[j].counterAxisSizingMode = 'AUTO'
+	}
+
+	// Comparing the row length of the selected table with the data
+	var rowLength = getTextNodes(selectedFrame, textNodes, columnLength)
+	if (rowLength != rowLengthData) {
+		if (resizable) {
+			updateRowCount(selectedFrame, columnLength, rowLength, rowLengthData)
+			textNodes = []
+			rowLength = getTextNodes(selectedFrame, textNodes, columnLength)
 		} else {
-		  newNode.x = original.x + original.width * (j + 1)
-		  newNode.y = original.y + original.height * (i + 1)
+			figma.closePlugin("🛑 Your given data has not the same row count as your selected table");
+			return
+		}
+
+	}
+
+	// Copy text from data cell into table
+	insertData(textNodes, cellsData, right_align_numbers)
+
+}
+
+function getTextNodes(selectedFrame: FrameNode, textFrames: TextNode[], columnLength: number) {
+	var maxRowLength = 0
+
+	if ("children" in selectedFrame) {
+		for (var i = 0; i < columnLength; i++) {
+			var column = selectedFrame.children[i] as FrameNode
+			var columnChildren = column.children
+			for (var j = 0; j < columnChildren.length; j++) {
+				if (maxRowLength < columnChildren.length) {
+					maxRowLength = columnChildren.length /* Getting the max row length */
+				}
+				var textFrame = getFirstChildOfTypeText(columnChildren[j])
+				if (textFrame != null && textFrame.type == 'TEXT') {
+					textFrames.push(textFrame)
+				}
+			}
 		}
+	}
+
+	return maxRowLength
+}
+
+function updateColumnCount(selectedFrame: FrameNode, columnLength: number, columnLengthData: number) {
```

Comment:
```suggestion
function addOrRemoveColumns(selectedFrame: FrameNode, columnLength: number, columnLengthData: number) {
```

Reply:
I will update this in my next commit so that when the function is called, it is also named correctly.